megamek/src/megamek/common/ICarryable.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package megamek.common;
+
+/**
+ * An interface defining all the required properties of a carryable object.
+ */
+public interface ICarryable {

```

Comment:
I had the hope at some point for something like this to be placed under (i.e. extend) BTObject to be in a neat hierarchy (fwiw). It would then be the first place to add this override:
```
default boolean isCarryableObject() {
    return true;
}
```
It could add a few overrides to return false to the other checks else but that's not terribly important.
It would make sense to change getName() to generalName() (I was out of ideas for a better term) which makes it the equivalent of getChassis().

(Also... it could be named Carryable without the I as there's no old Carryable class in the way)

Reply:
Didn't realize that there was a BTObject. I'll investigate how practical it'll be to update the interface to fit into the hierarchy.

The I in front of the name follows the general convention (and in MM as well to some extent) for interfaces; it's easier to tell at a glance that a piece of code is an interface if it has the I in front of it. Although I guess with the "default" functions nowadays the difference is a bit blurry.